### PR TITLE
Fixes the anti-drop implant not removing the nodrop trait on held items if they're somehow dropped.

### LINKED ISSUE #40669: The nodrop implant gives the items in your hands nodrop rather then making the person that has it unable to drop items
Title, give someone the nodrop implant, put something in their hands, make them turn on the implant, cut off the hand with a item in it, pick up the item, you are unable to drop it, 
fuck i worded the name of this badly.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. I have used comsigs to do it. Also I have improved the logic of its `ui_action_click`.

## Why It's Good For The Game
This will fix #40669.

## Changelog
:cl:
fix: Fixed the anti-drop implant not removing the nodrop trait on held items if they're somehow dropped.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/augments_internal.dm [hunk 1]
```diff
@@ -50,19 +50,16 @@
...
-			for(var/obj/item/stored_item in stored_items)
-				to_chat(owner, span_notice("Your [owner.get_held_index_name(owner.get_held_index_of_item(stored_item))]'s grip tightens."))
-				ADD_TRAIT(stored_item, TRAIT_NODROP, IMPLANT_TRAIT)
-
+			return
+		for(var/obj/item/held_item in owner.held_items)
+			stored_items += held_item
+			to_chat(owner, span_notice("Your [owner.get_held_index_name(owner.get_held_index_of_item(held_item))]'s grip tightens."))
+			ADD_TRAIT(held_item, TRAIT_NODROP, IMPLANT_TRAIT)
+			RegisterSignal(held_item, COMSIG_ITEM_DROPPED, .proc/on_held_item_dropped)
```
Comment (Watermelon914) [inline code/modules/surgery/organs/augments_internal.dm:62]: held_item can be null here.

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/augments_internal.dm [hunk 2]
```diff
@@ -50,19 +50,16 @@
...
 			to_chat(owner, span_notice("You are not holding any items, your hands relax..."))
 			active = FALSE
-			stored_items = list()
-		else
-			for(var/obj/item/stored_item in stored_items)
-				to_chat(owner, span_notice("Your [owner.get_held_index_name(owner.get_held_index_of_item(stored_item))]'s grip tightens."))
-				ADD_TRAIT(stored_item, TRAIT_NODROP, IMPLANT_TRAIT)
-
+			return
+		for(var/obj/item/held_item in owner.held_items)
```
[the remark this answers is not in the document]
Comment (Ghommie) [reply code/modules/surgery/organs/augments_internal.dm:58]: No. `held_items` may contain null entries. Basically the length of held_items is more or less the amount of hand slots a mob has.

[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/surgery/organs/augments_internal.dm:58]: Yeah but it's easier to check for null than to typecheck an object.

File: code/modules/surgery/organs/augments_internal.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/surgery/organs/augments_internal.dm:62]: Ok, wait. If you already knew `held_item` could be null if I implemented the above suggestion, why did you suggest it anyway?

**Pull request merged**